RDK-50163 Support for running Android containers

jonhenderson-sky opened the pull request:
- Manage mounting android partitions
- OCI Template modifications for Android
- Extra settings for Android specific bind mounts and devices

It is not required to modify the base dobby.bb, but a device specific addition to the Dobby config must be made as required to support running Android on a specific device.
 
### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (doesn't fit into the above categories - e.g. documentation updates)

### Requires Bitbake Recipe changes?
- [ ] The base Bitbake recipe (`meta-rdk-ext/recipes-containers/dobby/dobby.bb`) must be modified to support the changes in this PR (beyond updating `SRC_REV`)

jonhenderson-sky committed:
Support for running Android containers

- Manage mounting android partitions
- OCI Template modifications for Android
- Extra settings for Android specific bind mounts and devices

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/Dobby?pullRequest=327) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/Dobby?pullRequest=327) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/Dobby?pullRequest=327) it.</sub>

jonhenderson-sky committed:
Add extra masked paths

Signed-off-by: Jon Henderson <jon.henderson2@sky.uk>

jonhenderson-sky committed:
Add Android AppArmor profile setting

Provide the ability to use an alternate AppArmor profile when running
android containers, i.e.

{
    "android": {
        "appArmorProfile": "aosp_default"
    }
}

If no setting is provided, the normal Dobby profile will be used.

goruklu commented on CMakeLists.txt:
```diff
@@ -373,6 +374,10 @@ if(PLUGIN_GAMEPAD)
 endif()
 
+if(PLUGIN_ANDROIDRUNTIME)
+    add_subdirectory(rdkPlugins/AndroidRuntime)
```
I don't see the rdkPlugins/AndroidRuntime directory in this PR

goruklu commented:
@jonhenderson-sky The unit tests  and builds are failing, can you please correct those ?
Also, It would be great if you can add some unit tests for the Android container use case

jonhenderson-sky committed:
Added missing rdkPlugins directory

jonhenderson-sky committed:
Q3 Update

Support single compressed system.img (sqfs or erofs)
Remove special kernel commandline handling

jon-henderson-comcast committed:
Fix L1 test build
